cirq/contrib/routing/multi_prog_mapping.py

```diff
@@ -0,0 +1,696 @@
+
+            flayers = self.generate_front_layers(dags)
+
+        return schedule
+
+
+############################################################
+
+
+def multi_prog_map(device_graph: nx.Graph, single_er: Dict[Tuple[ops.Qid,],
+                                                           List[float]],
+                   two_er: Dict[Tuple[ops.Qid, ops.Qid], List[float]],
+                   prog_circuits: List[circuits.Circuit]) -> None:
+    twoQ_gate_type = cirq.CZ
+    treeObj = HierarchyTree(device_graph, single_er, two_er)
+    tree = treeObj.tree_construction()
+    print("tree")
+    print(tree.nodes)
+
+    parObj = QubitsPartitioning(tree, prog_circuits, single_er, two_er,
+                                twoQ_gate_type)
+    desc_cirs = parObj.circuits_descending()
+
+    partitions = parObj.qubits_allocation(desc_cirs)
+
+    print("partitions:")
+    print(partitions)
+
+    xswap = XSWAP(device_graph, desc_cirs, partitions, twoQ_gate_type)
+    schedule = xswap.insert_SWAP_and_generate_schedule()
+    print("schedule:")
+    print(schedule)
+
+
+def prepare_couplingGraph_errorValues(device_graph):
+    #single_er = load_calibrations()['single_qubit_p00_error'] # to do ??
+    #two_er = load_calibrations()['two_qubit_sycamore_gate_xeb_cycle_purity_error'] # to do ??
+
+    single_er = {
+        (cirq.GridQubit(1, 0),): [0.028600441075128205],
+        (cirq.GridQubit(0, 0),): [0.01138359559038841],
+        (cirq.GridQubit(1, 1),): [0.05313138858345922],
+        (cirq.GridQubit(0, 1),): [0.0005880214404983153],
+        (cirq.GridQubit(1, 2),): [0.0018232495924263727],
+        (cirq.GridQubit(0, 2),): [0.039571298178797366]
+    }
+    two_er = {
+        (cirq.GridQubit(1, 0), cirq.GridQubit(0, 0)): [0.018600441075128205],
+        (cirq.GridQubit(0, 0), cirq.GridQubit(0, 1)): [0.01938359559038841],
+        (cirq.GridQubit(1, 1), cirq.GridQubit(0, 1)): [0.01313138858345922],
+        (cirq.GridQubit(0, 1), cirq.GridQubit(0, 2)): [0.005880214404983153],
+        (cirq.GridQubit(1, 1), cirq.GridQubit(1, 2)): [0.008232495924263727],
+        (cirq.GridQubit(0, 2), cirq.GridQubit(1, 2)): [0.03571298178797366]
+    }
+
+    # coupling graph
+    dgraph = nx.Graph()
+
+    for q0, q1 in two_er:  # .items()
+        dgraph.add_edge(q0, q1)
+
+    # list of program circuits
+    qubits = cirq.LineQubit.range(3)
+    circuit1 = cirq.Circuit(cirq.X(qubits[0]), cirq.Y(qubits[1]),
+                            cirq.CZ(qubits[0], qubits[1]),
+                            cirq.CZ(qubits[1], qubits[2]))
+    qubits = cirq.LineQubit.range(2)
+    circuit2 = cirq.Circuit(cirq.X(qubits[0]), cirq.Y(qubits[1]),
+                            cirq.CZ(qubits[0], qubits[1]))
+    program_circuits = []
+    program_circuits.append(circuit2)
+    program_circuits.append(circuit1)
+
+    print(circuit2)
+    print(circuit1)
+
+    multi_prog_map(dgraph, single_er, two_er, program_circuits)
+
+
+if __name__ == "__main__":
```

Comment:
If this is meant to be executed as a standalone, I would include an example command line at the top of the file.